hazelcast/src/main/java/com/hazelcast/internal/metrics/managementcenter/MetricsCompressor.java

```diff
@@ -87,8 +84,8 @@ public void addLong(String name, long value) {
     public void addDouble(String name, double value) {
         try {
             writeName(name);
-            // convert to long with specified precision
-            dos.writeLong(Math.round(value * DOUBLE_TO_LONG));
+            dos.writeByte(ValueType.DOUBLE.ordinal());
```

Comment:
if we really need types, then 1 byte would be enough.

Reply:
The type is needed for MC. We change `long` -> `byte`+`long/double`, so we add only 1 byte for the type. Maybe I miss your point?

Reply:
ah ok, I missed the writeByte() since ordinal is int, wouldn't it be better to be more explicit rather than doing downcasting?

Reply:
> ah ok, I missed the writeByte() since ordinal is int, wouldn't it be better to be more explicit rather than doing downcasting?

Signature of `DataOutputStream#writeByte` is `(int)`. So, even if we pass a `byte`, there will be an implicit type conversion. The value of using `ValueType#ordinal` is that it provides fast lookups during deserialization: `ValueType#valueOf(int)` simply accesses an array via index. That's why I didn't go with a separate `byte` field in `ValueType`. Also using `ValueType#ordinal` is more or less safe here, as `MetricsServiceTest` covers this logic.

We already have examples of such downcasting in this class. See `#writeName`:
```
dos.writeShort(equalPrefixLength); // here `equalPrefixLength` is `int`
```

WDYT?